lib/AST/Module.cpp

```diff
@@ -901,6 +901,27 @@ ModuleDecl *ModuleDecl::getTopLevelModule(bool overlay) {
   return this;
 }
 
+bool ModuleDecl::isSubmoduleOf(const ModuleDecl *M) const {
+  // Swift modules don't currently support submodules.
+  if (!isNonSwiftModule())
+    return false;
+
+  auto *ClangParent = M->findUnderlyingClangModule();
+  if (!ClangParent)
+    return false;
+
+  auto *ClangModule = findUnderlyingClangModule();
+  if (!ClangModule)
+    return false;
+
+  while ((ClangModule = ClangModule->Parent)) {
```

Comment:
```suggestion
ClangModule->isSubmoduleOf(ClangParent)
```

Reply:
This is the implementation of `isSubmoduleOf`, and this is a single-equals assignment anyway; i'm checking that `ClangModule->Parent` exists, and also assigning it to `ClangModule` for the next loop iteration.

Reply:
I was referring to using this https://clang.llvm.org/doxygen/classclang_1_1Module.html#ab9ffc11ef69081f60cdc8fabe67f1cbf but maybe I got confused by which types are swift modules vs clang modules.

Reply:
Oh! I totally missed that there was an `isSubmoduleOf` on Clang modules. I can replace the entire loop with that call, then.